WooCommerce/TodayStatsWidget/TodayViewController.swift

```diff
@@ -0,0 +1,366 @@
+        dateFormatter.timeZone = site.siteTimezone
+        let earliestDate = dateFormatter.string(from: Date().startOfDay(timezone: site.siteTimezone))
+        let latestDate = dateFormatter.string(from: Date().endOfDay(timezone: site.siteTimezone))
+
+        let quantity = timeRange.siteVisitStatsQuantity(date: Date(), siteTimezone: site.siteTimezone)
+
+        let group = DispatchGroup()
+
+        /// Load Order Stats
+        group.enter()
+        let remoteOrderStats = OrderStatsRemoteV4(network: network)
+        remoteOrderStats.loadOrderStats(for: site.siteID,
+                                        unit: timeRange.intervalGranularity,
+                                        earliestDateToInclude: earliestDate,
+                                        latestDateToInclude: latestDate,
+                                        quantity: quantity) { [weak self] (orderStatsV4, error) in
+            guard error == nil else {
+                if let error = error as? DotcomError, error == .noRestRoute {
+                    self?.error = .statsV4NotAvailable
+                }
+                else {
+                    self?.error = .errorFetchingStats
+                }
+                group.leave()
+                return
+            }
+            tempStats.totalOrders = orderStatsV4?.totals.totalOrders
+
+            // TODO: implement currency formatter
+            // let currencyCode = CurrencySettings.shared.symbol(from: CurrencySettings.shared.currencyCode)
+            // totalRevenueText = CurrencyFormatter().formatHumanReadableAmount(String("\(orderStats.totals.grossRevenue)"), with: currencyCode) ?? String()
+
+            tempStats.totalRevenue = orderStatsV4?.totals.grossRevenue
+            group.leave()
+        }
+
+        /// Load Visit Stats
+        group.enter()
+        let remoteVisitStats = SiteVisitStatsRemote(network: network)
+        remoteVisitStats.loadSiteVisitorStats(for: site.siteID,
+                                              siteTimezone: site.siteTimezone,
+                                    unit: timeRange.siteVisitStatsGranularity,
+                                    latestDateToInclude: Date().endOfDay(timezone: site.siteTimezone),
+                                    quantity: quantity) { [weak self] (siteVisitStats, error) in
+                                        guard error == nil else {
+                                            self?.error = .errorFetchingStats
+                                            group.leave()
+                                            return
+                                        }
+
+                                        tempStats.totalVisitors = siteVisitStats?.totalVisitors
+                                        group.leave()
+        }
+
+        group.notify(queue: .main) { [weak self] in
+            if tempStats.isVisitStatsEmpty() || tempStats.isOrderStatsEmpty() {
```

Comment:
nit: these two conditions feel like lower level details from the same class `StatsData`, how about grouping these two to a function in `StatsData` like `hasData()`?